src/main/java/updater/UpdateManager.java

```diff
@@ -0,0 +1,151 @@
+
+    private static final String UPDATE_DIRECTORY = "updates";
+
+    private static final String UPDATE_SERVER_DATA_NAME =
+            "https://raw.githubusercontent.com/HubTurbo/AutoUpdater/master/HubTurbo.xml";
+    private static final String UPDATE_LOCAL_DATA_NAME = "HubTurbo.json";
+
+    private static final String UPDATE_APP_NAME = "HubTurbo.jar";
+
+    private final FileDownloader fileDownloader;
+
+    private final ExecutorService pool = Executors.newSingleThreadExecutor();
+
+    public UpdateManager() {
+        fileDownloader = new FileDownloader();
+    }
+
+    /**
+     * Driver method to trigger UpdateManager to run. Update will be run on another thread.
+     *
+     * - Run is not automatic upon instancing the class in case there would like to be conditions on when to run update,
+     *   e.g. only if user is logged in
+     */
+    public void run() {
+        if (!TestController.isTestMode()) {
+            pool.execute(() -> runUpdate());
+        }
+    }
+
+    /**
+     * Method that will run update sequence.
+     * 1. Initialize system for update, e.g. make directory
+     * 2. Check if there is any new update
+     *   2.a. If there is no new update but the previously downloaded update is not applied yet,
+     *        prompt user for updating and restarting application
+     * 3. Download update according to user preference
+     * 4. Once update is done, prompt user for updating and restarting application
+     */
+    private void runUpdate() {
+        if (!initUpdate()) {
+            logger.error("Failed to initialize update");
+        }
+
+        downloadUpdateData();
+
+        // TODO check if there is a new update since last update, and download update if necessary
+
+        downloadUpdateForApplication();
+
+        // TODO once update is done, run new process that will:
+        // - kill current HubTurbo
+        // - replace JAR
+        // - start the new JAR
+
+    }
+
+    /**
+     * Initializing system for updates
+     * - Create directory(ies) for updates
+     */
+    private boolean initUpdate() {
+        boolean result = false;
+
+        File updateDir = new File(UPDATE_DIRECTORY);
+
+        if (!updateDir.exists()) {
+            result = updateDir.mkdirs();
+            if (!result) {
+                logger.error("Failed to create update directories");
+            }
+        }
+
+        return result;
+    }
+
+    private void downloadUpdateData() {
+        try {
+            fileDownloader.setProgressTrackerWindowDescription("Checking for update", "Checking for update...");
+            fileDownloader.setEnableProgressWindow(false);
+            fileDownloader.downloadFileFromUrlToFile(
+                    new URI(UPDATE_SERVER_DATA_NAME),
+                    new File(UPDATE_DIRECTORY + File.separator + UPDATE_LOCAL_DATA_NAME));
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
```

Comment:
We should log using a logger, so the output goes into the same place. Same for below.